python/lsst/daf/butler/script/register_dataset_type.py

```diff
@@ -0,0 +1,76 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("register_dataset_type",)
+
+from typing import Tuple
+
+from .. import Butler
+from ..core import DatasetType
+
+
+def register_dataset_type(repo: str, dataset_type: str, storage_class: str,
+                          dimensions: Tuple[str, ...], is_calibration: bool = False) -> bool:
+    """Register a new dataset type.
+
+    Parameters
+    ----------
+    repo : `str`
+        URI string of the Butler repo to use.
+    dataset_type : `str`
+        The name of the new dataset type.
+    storage_class : `str`
+        The name of the storage class associated with this dataset type.
+    dimensions : `tuple` of `str`
```

Comment:
Is the butler standard for documenting sequence types different from the lsst numpydoc one?

```
`list` [`str`]
```

https://developer.lsst.io/python/numpydoc.html#sequence-types

Reply:
The convention is mostly desperately waiting for sphinx to catch up so that we can delete it completely (since we use type annotations almost everywhere). I'll update it.